src/program/snabblwaftr/run/run.lua

```diff
@@ -0,0 +1,149 @@
+
+   -- It's essential to initialize the binding table before the aftrconf
+   bt.get_binding_table(bt_file)
+   local aftrconf = conf.get_aftrconf(conf_file)
+
+   local c = config.new()
+   config.app(c, 'inetNic', Intel82599, {pciaddr=inet_nic_pci,
+                                         macaddr = ethernet:ntop(aftrconf.aftr_mac_inet_side)})
+   config.app(c, 'b4sideNic', Intel82599, {pciaddr=b4side_nic_pci,
+                                           macaddr = ethernet:ntop(aftrconf.aftr_mac_b4_side)})
+   config.app(c, 'lwaftr', lwaftr.LwAftr, aftrconf)
+   config.app(c, 'v6_stats', basic_apps.Statistics)
+   config.app(c, 'v4_stats', basic_apps.Statistics)
+
+   config.link(c, 'inetNic.tx -> lwaftr.v4')
+   config.link(c, 'b4sideNic.tx -> lwaftr.v6')
+   if verbose then
+      config.link(c, 'lwaftr.v4 -> v4_stats.input')
+      config.link(c, 'v4_stats.output -> inetNic.rx')
+      config.link(c, 'lwaftr.v6 -> v6_stats.input')
+      config.link(c, 'v6_stats.output -> b4sideNic.rx')
+   else
+      config.link(c, 'lwaftr.v4 -> inetNic.rx')
+      config.link(c, 'lwaftr.v6 -> b4sideNic.rx')
+   end
+
+   engine.configure(c)
+
+   if ultra_verbose then
+      local function lnicui_info() print_info(engine) end
+      local t = timer.new("report", lnicui_info, 1e9, 'repeating')
+      timer.activate(t)
+   end
+
+   if duration then
+      engine.main({duration=duration, report={showlinks=true}})
+   else
+      engine.main({report={showlinks=true}})
+   end
+end
+
+function run (parameters)
+   if #parameters == 0 then
+      usage()
+      main.exit(1)
+   end
+   local opt = {}
+   function opt.v (arg) verbose = true  end
+   function opt.u (arg) ultra_verbose = true verbose = true end
+   function opt.D (arg) duration = tonumber(arg)  end
+   function opt.b (arg) bt_file = arg end
+   function opt.c (arg) conf_file = arg end
+   function opt.n (arg) v4_pci = arg end
+   function opt.m (arg) v6_pci = arg end
+   function opt.h () usage() main.exit(1) end
+   lib.dogetopt(parameters, opt, "b:c:n:m:vuDh", long_opts)
+
+   -- Check required arguments
+   if not bt_file then
+      fatal("Argument '--bt' was not set")
+   end
+   if not conf_file then
+      fatal("Argument '--conf' was not set")
+   end
+   if not v4_pci then
+      fatal("Argument '--v4-pci' was not set")
+   end
+   if not v6_pci then
+      fatal("Argument '--v6-pci' was not set")
+   end
+
+   -- Check arguments are valid
+   if not file_exists(bt_file) then
+      fatal(("Couldn't locate binding-table at %s"):format(bt_file))
+   end
+   if not file_exists(conf_file) then
+      fatal(("Couldn't locate configuration file at %s"):format(conf_file))
+   end
+   if not nic_exists(v4_pci) then
+      fatal(("Couldn't locate NIC with PCI address '%s'"):format(v4_pci))
+   end
+   if not nic_exists(v6_pci) then
+      fatal(("Couldn't locate NIC with PCI address '%s'"):format(v6_pci))
+   end
+
+   -- Run the main program
+   nic_ui({ bt_file, conf_file, v4_pci, v6_pci })
```

Comment:
Probably I would do this instead:
```
nic_ui(bt_file, conf_file, v4_pci, v6_pci)
```
…and avoid the call to `unpack()` inside the `nic_ui()` function.

If you think the number of argument is going to grow much later, it may make sense to keep the parameters as a table, though. In that case, I think it's not a big deal to leave the table here.